module/VuFind/src/VuFind/Config/PathResolverFactory.php

```diff
@@ -90,58 +85,24 @@ public function __invoke(
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
-        $localDirs = [];
-        $currentDir = defined('LOCAL_OVERRIDE_DIR')
-            && strlen(trim(LOCAL_OVERRIDE_DIR)) > 0
-            ? LOCAL_OVERRIDE_DIR : '';
-        while (!empty($currentDir)) {
-            // check if the directory exists
-            if (!($canonicalizedCurrentDir = realpath($currentDir))) {
-                trigger_error('Configured local directory does not exist: ' . $currentDir, E_USER_WARNING);
-                break;
-            }
-            $currentDir = $canonicalizedCurrentDir;
-
-            // check if the current directory was already included in the stack to avoid infinite loops
-            if (in_array($currentDir, array_column($localDirs, 'directory'))) {
-                trigger_error('Current directory was already included in the stack: ' . $currentDir, E_USER_WARNING);
-                break;
-            }
-
-            // loading DirLocations.ini of currentDir
-            $systemConfigFile = $currentDir . '/DirLocations.ini';
-            $systemConfig = new Config(
-                file_exists($systemConfigFile)
-                    ? $this->getIniReader()->fromFile($systemConfigFile)
-                    : []
-            );
-
-            // adding directory to the stack
-            array_unshift(
-                $localDirs,
-                [
-                    'directory' => $currentDir,
+        $dirLocationsStack = $this->getDirLocationsStack();
+        $localDirStack = array_map(
+            function ($dirLocations) {
```

Comment:
As noted elsewhere, I think singular $dirLocation might be more clear here.

Reply:
Due to other changes this became obsolete.